### PR TITLE
ci: sdk binary size report

### PR DESCRIPTION
**Pull request by mrehan27**
helps: https://linear.app/customerio/issue/MBL-157/track-the-binary-size-of-our-android-sdk

### Changes

- Added workflow to comment SDK binary size change summary on every PR
- Updated workflow to commit SDK size report for every commit merged to `main` before being released

### CODE REVIEW

File: .github/workflows/deploy-sdk.yml [hunk 1]
```diff
@@ -25,6 +25,12 @@ jobs:
         # uses: kenji-miyake/setup-sd@v1
         uses: levibostian/setup-sd@add-file-extension # Using fork until upstream Action has bug fixed in it.
 
+      - name: Generate SDK Size Report
+        run: ./gradlew generateSdkSizeReport -PsaveReportTo=reports
+
+      - name: Stage SDK Size Report
+        run: git add reports/
+
```
Comment (mrehan27) [inline .github/workflows/deploy-sdk.yml:33]: I haven't been able to test this as this only runs when we prepare for release. So would appreciate more eyes on it.

About the workflow, the change is to create and store report in `main` branch whenever a release is made. [Added more details about it on Linear](https://linear.app/customerio/issue/MBL-157/track-the-binary-size-of-our-android-sdk#comment-5bc03dee).

**Comment by mrehan27** [comment]
[Test PR](https://github.com/customerio/customerio-android/pull/292#issuecomment-2024878264) to verify changes for adding comment when binary size changes.

File: .github/workflows/sdk-binary-size.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        with:
+          ref: ${{ github.base_ref }}
+
+      - name: Generate report for base branch
+        run: |
+          ./gradlew publishToMavenLocal
+          ./gradlew generateSdkSizeReport
+        env:
+          # Use local version to make sure the report is generated for the current changes
+          MODULE_VERSION: 'local'
```
Comment (Shahroz16) [inline .github/workflows/sdk-binary-size.yml:26]: if we have `IS_DEVELOPMENT` true, wouldn't it automatically set the version to local?

Comment (mrehan27) [reply .github/workflows/sdk-binary-size.yml:26]: Looking at the code, this seems correct. I haven't tried it yet, will update and see if it works fine 👍🏻

**Review by Shahroz16 — APPROVED** [review]

File: .github/workflows/deploy-sdk.yml (hunk 1 above)
Comment (Shahroz16) [reply .github/workflows/deploy-sdk.yml:33]: I would rather suggest move the `Stage SDK Size Report` part in the Semantic assets section we have, we already stage files in it.

**Pull request merged**